app/main/views/briefs.py

```diff
@@ -174,19 +174,24 @@ def update_brief_status(brief_id, action):
     brief = Brief.query.filter(
         Brief.id == brief_id
     ).first_or_404()
-
-    if brief.framework.status != 'live':
-        abort(400, "Framework is not live")
+    framework_status = brief.framework.status
 
     action_to_status = {
         'publish': 'live',
         'withdraw': 'withdrawn',
         'cancel': 'cancelled',
         'unsuccessful': 'unsuccessful'
     }
-    if brief.status != action_to_status[action]:
-        previousStatus = brief.status
-        brief.status = action_to_status[action]
+    new_status = action_to_status[action]
+
+    if new_status in ['live', 'withdrawn'] and framework_status != 'live':
+        abort(400, "Framework is not live")
+    if new_status in ['cancelled', 'unsuccessful'] and framework_status not in ['live', 'expired']:
```

Comment:
I don't think we need this second check. If the framework is not live or expired then it will not be in a state where you could ever create a brief, let alone publish one

Reply:
Yeah it will never be `closed` if the framework isn't `live` or `expired`